src/lib/statsig.ts

```diff
@@ -0,0 +1,50 @@
+import { StatSigFlags, StatsigConfigType } from '@/types/statsig';
+import { StatsigClient } from '@statsig/js-client';
+
+const fetchIp = async () => {
+  const resp = await fetch('https://api.ipify.org?format=json');
+  return (await resp.json())?.ip;
+};
+
+let statsigClient: StatsigClient;
+
+const initStatsig = async () => {
+  if (statsigClient) return statsigClient;
+  statsigClient = new StatsigClient(
+    import.meta.env.VITE_STATSIG_CLIENT_KEY,
+    {
+      ip: await fetchIp(),
+    },
+    {
+      disableLogging: import.meta.env.VITE_DISABLE_STATSIG,
+      disableStorage: import.meta.env.VITE_DISABLE_STATSIG,
+    }
+  );
+  if (import.meta.env.VITE_DISABLE_STATSIG) return statsigClient;
+  await statsigClient.initializeSync();
+  return statsigClient;
+};
+
+export const statsigClientPromise = initStatsig();
+
+const statsigCheckGatePromise = async (gateId: StatSigFlags) => {
+  const client = await statsigClientPromise;
+  return client.checkGate(gateId);
+};
+
+/**
+ * This function should typically be avoided in favor of the useStatsig hook
+ * This is used in useInitializePage to retrieve configs for abacus only once
+ * inside of a useEffect hook.
+ *
+ */
+export const statsigGetAllGateValuesPromise = async () => {
+  const results = await Promise.all(
+    Object.values(StatSigFlags).map(async (gate) => ({
+      [gate]: await statsigCheckGatePromise(gate),
```

Comment:
hmm do we want to run `await statsigClientPromise` on every gate? I think it might be easier to just call `client = initStatsig` at the top of this function, and then call `client.checkGate()` on each gate

Reply:
hmm maybe. originally we wanted this because we were calling this all within the abacus constructor. lemme check